jupyterhub_config.py

```diff
@@ -13,6 +13,12 @@
 c.Spawner.environment = {
     "SPARKMAGIC_CONF_DIR": os.environ.get("SPARKMAGIC_CONF_DIR", "~/.sparkmagic"),
     "JUPYTER_ENABLE_LAB": "yes",
+    "HTTP_PROXY": os.environ.get("HTTP_PROXY"),
+    "HTTPS_PROXY": os.environ.get("HTTPS_PROXY"),
+    "NO_PROXY": os.environ.get("NO_PROXY"),
+    "http_proxy": os.environ.get("http_proxy"),
```

Comment:
Looks ok, we could have mapped both lower and uppercase values to the same os.environ.get() string, but keeping them matching case is ok.